src/openMVG/sfm/pipelines/sfm_regions_provider.hpp

```diff
@@ -88,20 +88,25 @@ struct Regions_Provider
   virtual bool load(
     const SfM_Data & sfm_data,
     const std::string & feat_directory,
-    std::unique_ptr<features::Regions>& region_type)
+    std::unique_ptr<features::Regions>& region_type,
+    C_Progress& my_progress_bar = C_Progress::dummy())
   {
     region_type_.reset(region_type->EmptyClone());
 
-    C_Progress_display my_progress_bar( sfm_data.GetViews().size(),
-      std::cout, "\n- Regions Loading -\n");
+    my_progress_bar.restart(sfm_data.GetViews().size(), "\n- Regions Loading -\n");
     // Read for each view the corresponding regions and store them
-    bool bContinue = true;
+    volatile bool bContinue = true;
```

Comment:
I could go either way. The code as is should do what it's supposed to since 'volatile' guarantees that bContinue is actually evaluated each time and there is no way a bool enters an invalid state. On the other hand - the cost of one atomic access is neglible compared to reading a feature file. You decide.